pypeit/core/telluric.py

```diff
@@ -2530,7 +2532,16 @@ def run(self, only_orders=None):
         only_orders = [only_orders] if (only_orders is not None and
                                         isinstance(only_orders, (int, np.integer))) \
                                     else only_orders
-        good_orders = self.srt_order_tell if only_orders is None else only_orders
+        # by default, we use all orders
+        good_orders = self.srt_order_tell
+        # If only_orders is not None, then we check that the orders are in the data
+        if only_orders is not None and self.ech_orders is not None and len(self.ech_orders) == self.norders:
+            if np.where(np.isin(self.ech_orders,only_orders))[0].size == 0:
```

Comment:
Nice!  Two recommendations:

Get the indices in `ech_orders` once:
```
indx = np.where(np.isin(self.ech_orders, only_order))[0]
if indx.size == 0:
    ...
else:
   good_orders = indx
```

Check that all of the orders selected by `only_orders` were found.  Something like:
```
if len(indx) != len(only_orders):
    missing_orders = list(set(only_orders) - set(self.ech_orders[indx])) 
    msgs.warn(f'Some orders provided in `only_orders` are not among the expected orders.  Ignoring orders: {missing_orders}')